spock-core/src/main/java/spock/lang/Retry.java

```diff
@@ -43,6 +44,18 @@
    */
   Class<? extends Throwable>[] exceptions() default {Exception.class, AssertionError.class};
 
+  /**
+   * Condition that is evaluated to decide whether the feature should be
+   * retried.
+   *
+   * The feature is retried if the exception class passes the type check and the
+   * specified condition holds true. If no condition is specified, only the type
+   * check is performed.
+   *
```

Comment:
you should mention either the RetryConditionContext or the available variables in the docs. I just noticed that this information is also missing in `@Requires`/`@IgnoreIf`, maybe you could add it there as well.